instrumentation-api-semconv/src/main/java/io/opentelemetry/instrumentation/api/instrumenter/net/PeerServiceResolverImpl.java

```diff
@@ -0,0 +1,115 @@
+/*
+ * Copyright The OpenTelemetry Authors
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+package io.opentelemetry.instrumentation.api.instrumenter.net;
+
+import static java.util.Comparator.comparing;
+import static java.util.Comparator.naturalOrder;
+import static java.util.Comparator.nullsFirst;
+
+import java.io.Serializable;
+import java.util.Comparator;
+import java.util.HashMap;
+import java.util.Map;
+import java.util.Objects;
+import javax.annotation.Nullable;
+
+class PeerServiceResolverImpl implements PeerServiceResolver {
+
+  private static final Comparator<ServiceMatcher> matcherComparator =
+      nullsFirst(
+          comparing(ServiceMatcher::getPort, nullsFirst(naturalOrder()))
+              .thenComparing(comparing(ServiceMatcher::getPath, nullsFirst(naturalOrder()))));
+
+  private final Map<String, Map<ServiceMatcher, String>> mapping = new HashMap<>();
+
+  PeerServiceResolverImpl(Map<String, String> peerServiceMapping) {
+    peerServiceMapping.forEach(
+        (key, serviceName) -> {
+          String url = key.contains("://") ? key : ("http://" + key);
+          String host = UrlParser.getHost(url);
+          Integer port = UrlParser.getPort(url);
+          String path = UrlParser.getPath(url);
+          Map<ServiceMatcher, String> matchers =
+              mapping.computeIfAbsent(host, x -> new HashMap<>());
+          matchers.putIfAbsent(new ServiceMatcher(port, path), serviceName);
+        });
+  }
+
+  @Override
+  public boolean isEmpty() {
+    return mapping.isEmpty();
+  }
+
+  @Override
+  public String resolveService(String host, Integer port, String path) {
+    Map<ServiceMatcher, String> matchers = mapping.get(host);
+    if (matchers == null) {
+      return null;
+    }
+    return matchers.entrySet().stream()
+        .filter(entry -> entry.getKey().matches(port, path))
+        .max((o1, o2) -> matcherComparator.compare(o1.getKey(), o2.getKey()))
+        .map(Map.Entry::getValue)
+        .orElse(null);
+  }
+
+  private static class ServiceMatcher implements Serializable {
```

Comment:
does it need to be Serializable?

might be nice to use `@AutoValue` for this class